File: 223.RectangleArea/MarkCSIE/RectangleArea/src/RectangleArea.java

public class RectangleArea {

    public static void main(String[] args) {
        RectangleArea test = new RectangleArea();
    }
    
        public int computeArea(int A, int B, int C, int D, int E, int F, int G, int H) {
        int areaA = (C - A) * (D - B);
        int areaB = (G - E) * (H - F);

        // no overlap
        if (A > G || E > C || B > H || F > D) {
            return areaA + areaB;
        } else {
            int right = Math.min(C, G);
            int left = Math.max(A, E);
            int top = Math.min(H, D);
            int bottom = Math.max(F, B);
            return areaA + areaB - (right - left) * (top - bottom);
        }
    }
}

File: 223.RectangleArea/Ddfish/RectangleArea.swift
class Solution {
    func computeArea(_ A: Int, _ B: Int, _ C: Int, _ D: Int, _ E: Int, _ F: Int, _ G: Int, _ H: Int) -> Int {
        
         var left = max(A,E)
         var right = max(min(C,G), left);
         var bottom = max(B,F)
         var top  = max(min(D,H), bottom);
        return (C-A)*(D-B) - (right-left)*(top-bottom) + (G-E)*(H-F);
    }
}
